### ISSUE TITLE
Failed to reproduce results in CIFAR10/100 (single augmentation)

### ISSUE THREAD

**Issue by ihounie** [MSU-MLSys-Lab/DeepAA#7]
Hi, as specified in the readme I tried to evaluate the augmentation policies by running:

python -m DeepAA_evaluate.train -c confs/wresnet28x10_cifar10_DeepAA_1.yaml --dataroot ./data --tag Exp_DeepAA_cifar10

And got the following results (mean of five runs):

|          | Acc   |
|----------|-------|
| CIFAR 10 |       |
| WRN40-2  | 96.37 |
| WRN28-10 | 97.44 |
| CIFAR100 |       |
| WRN40-2  | 79.30 |
| WRN28-10 | 83.64 |

Which is considerably lower than results reported in the paper for wresnet28-10.
I also evaluated on wresnet40-20, which is the architecture on which the search is performed.

Am I doing sth wrong?

**Comment by ihounie** [comment]
I have a theory of what may be causing this, if you run DeepAA_evaluate.train with --save and the same path five times, runs (after the first one) will start training from the previous trained model weights. This results in higher accuracy and is closer to reported results.

 Do you happen to have a script that reproduces the results of the paper?

**Comment by zhengyu998** [comment]
Hi,
Sorry for the late reply as I'm currently working on another project. 

Using the [second policy](https://github.com/MSU-MLSys-Lab/DeepAA/blob/master/policy_port/policy_DeepAA_cifar_2.npz) marked as `EXP: cifar_2` in the config with a slightly different weight decay should give you a better performance. You will be able to reproduce the results (or probably even better results than the ones reported in our paper) by replace `confs/wresnet28x10_cifarXX_DeepAA.yaml` with the following parameters.

For CIFAR-10 with WRN28-10, I've tested the following configuration for 4 runs which gives accuracy 97.58/97.57/97.62/97.77
```yml
model:
  type: wresnet28_10
dataset: cifar10
aug: DeepAA
deepaa:
  EXP: cifar_2
cutout: -1
batch: 128
gpus: 1
augmentation_search_space: Not_used # fixed_standard
epoch: 200
lr: 0.1
lr_schedule:
  type: 'cosine'
  warmup:
    multiplier: 1
    epoch: 5
optimizer:
  type: sgd
  nesterov: True
  decay: 0.00025
```

For CIFAR-100 with WRN28-10, I've tested the following configuration for 4 runs which gives accuracy 84.59/84.21/84.18/83.9
```yml
model:
  type: wresnet28_10
dataset: cifar100
aug: DeepAA
deepaa:
  EXP: cifar_2
cutout: -1
batch: 128
gpus: 1
augmentation_search_space: Not_used # fixed_standard
epoch: 200
lr: 0.1
lr_schedule:
  type: 'cosine'
  warmup:
    multiplier: 1
    epoch: 5
optimizer:
  type: sgd
  nesterov: True
  decay: 0.001
```

Also as you have noticed, please use different `--save` and `--tag` parameters for each independent run to prevent evaluating from other runs
```bash
python -m DeepAA_evaluate.train -c confs/wresnet28x10_cifarXX_DeepAA.yaml --dataroot ./data --save ckpt/diffenent_for_each_run.pth --tag different_for_each_run
```

**Comment by ihounie** [comment]
Thanks for the clarification and quick reply. Adjusting the  weight decay does improve performance (as already mentioned in the fast autoaugment paper)

I am a bit puzzled because afaik this is *not* the weight decay used in fast-autoaugment and other baselines, and  the paper states
" The evaluation configurations are kept consistent with that of Fast AutoAugment. "
